File: backend/src/main/java/com/incidentmanager/dto/response/IncidentResponse.java
package com.incidentmanager.dto.response;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import com.incidentmanager.enums.Priority;
import com.incidentmanager.enums.Status;

public record IncidentResponse(

    UUID id,
    String titulo,
    String descricao,
    Priority prioridade,
    Status status,
    String autor,
    String responsavelEmail,
    List<String> tags,
    LocalDateTime dataAbertura,
    LocalDateTime dataAtualizacao
) {
}

File: frontend/src/app/models/response/incident-response.ts
import { Priority } from '../enums/priority';
import { Status } from '../enums/status';

export interface IncidentResponse {
  id: string;
  titulo: string;
  descricao: string | null;
  prioridade: Priority;
  status: Status;
  autor: string;
  responsavelEmail: string;
  tags: string[];
  dataAbertura: string;
  dataAtualizacao: string;
}
